pilot/pkg/model/push_context.go

```diff
@@ -227,8 +223,8 @@ func (ps *PushContext) OnConfigChange() {
 // UpdateMetrics will update the prometheus metrics based on the
 // current status of the push.
 func (ps *PushContext) UpdateMetrics() {
-	ps.mutex.Lock()
-	defer ps.mutex.Unlock()
+	ps.proxyStatusMutex.RLock()
+	defer ps.proxyStatusMutex.RUnlock()
```

Comment:
this is a write lock..

Reply:
`mmap, f := ps.ProxyStatus[pm.Name]` isn't this a read lock?

Reply:
We just protect `ProxyStatus` with `proxyStatusMutex`.

Reply:
It’s reading that object and then the code is updating the metrics. As such metrics should not have any locks. It’s okay to get slightly inconsistent metrics instead of locks that hurt performance of the critical path.